MantidQt/CustomInterfaces/test/EnggDiffractionViewMock.h

```diff
@@ -105,7 +105,22 @@ class MockEnggDiffractionView
   MOCK_CONST_METHOD0(saveSettings, void());
 
   // virtual void plotFocusedSpectrum();
-  MOCK_METHOD1(plotFocusedSpectrum, void(const std::string&));
+  MOCK_METHOD1(plotFocusedSpectrum, void(const std::string &));
+
+  // void plotFocusStatus();
+  MOCK_METHOD0(plotFocusStatus, void());
+
+  // void plotRepChanged();
+  MOCK_METHOD1(plotRepChanged, void(int idx));
+
+  // virtual void plotWaterfallSpectrum
+  MOCK_METHOD1(plotWaterfallSpectrum, void(const std::string &wsName));
+
+  // virtual void plotReplacingWindow
+  MOCK_METHOD1(plotReplacingWindow, void(const std::string &wsName));
+
+  // std::string currentPlotType
+  MOCK_CONST_METHOD0(currentPlotType, int());
```

Comment:
Ideally this new functionality should be covered in the unit tests.
The positive cases are too slow to have them enabled, not everything can be run all the time, but:
- you can add a note in the disabled tests (there are 2 or 3 tests that have the `disabled_` prefix so they don't normally run.
- you can also add a EXPECT_CALL with 0 for the new plot methods in the negative test cases, just to make sure that calls to those methods are never introduced in the wrong places.